fdbclient/include/fdbclient/BulkLoading.h

```diff
@@ -541,64 +539,177 @@ enum class BulkLoadPhase : uint8_t {
 	Error = 6, // Updated by DD when this task has unretriable error
 };
 
+struct BulkLoadManifestSet {
+public:
+	constexpr static FileIdentifier file_identifier = 1384493;
+
+	BulkLoadManifestSet() = default;
+
+	BulkLoadManifestSet(int inputMaxCount) { maxCount = inputMaxCount; }
+
+	bool isValid() const {
+		if (maxCount == 0) {
+			return false;
+		}
+		if (manifests.empty()) {
+			return false;
+		}
+		if (manifests.size() > maxCount) {
+			return false;
+		}
+		if (!minBeginKey.present()) {
+			return false;
+		}
+		if (!maxEndKey.present()) {
+			return false;
+		}
+		for (const auto& manifest : manifests) {
+			if (!manifest.isValid()) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	// Return true if succeed
+	bool addManifest(const BulkLoadManifest& manifest) {
+		if (manifests.size() > maxCount) {
+			return false;
+		}
+		manifests.push_back(manifest);
+		if (transportMethod == BulkLoadTransportMethod::Invalid) {
+			transportMethod = manifest.getTransportMethod();
+		} else {
+			ASSERT(transportMethod == manifest.getTransportMethod());
+		}
+		if (loadType == BulkLoadType::Invalid) {
+			loadType = manifest.getLoadType();
+		} else {
+			ASSERT(loadType == manifest.getLoadType());
+		}
+		if (!byteSampleSetting.isValid()) {
+			byteSampleSetting = manifest.getByteSampleSetting();
+		} else {
+			ASSERT(byteSampleSetting == manifest.getByteSampleSetting());
+		}
+		if (!minBeginKey.present() || minBeginKey.get() > manifest.getBeginKey()) {
+			minBeginKey = manifest.getBeginKey();
+		}
+		if (!maxEndKey.present() || maxEndKey.get() < manifest.getEndKey()) {
+			maxEndKey = manifest.getEndKey();
+		}
+		return true;
+	}
+
+	bool isFull() const { return manifests.size() >= maxCount; }
+
+	bool hasEmptyData() const {
+		for (const auto& manifest : manifests) {
+			if (!manifest.hasEmptyData()) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	int64_t getTotalBytes() const {
+		int64_t res = 0;
+		for (const auto& manifest : manifests) {
+			res = res + manifest.getTotalBytes();
+		}
+		return res;
+	}
+
+	const std::vector<BulkLoadManifest>& getManifests() const { return manifests; }
+
+	BulkLoadTransportMethod getTransportMethod() const { return transportMethod; }
+
+	BulkLoadByteSampleSetting getByteSampleSetting() const { return byteSampleSetting; }
+
+	BulkLoadType getLoadType() const { return loadType; }
```

Comment:
Ditto

Reply:
I think it is fine because `isValid()` will check this before using it.

Reply:
I will add an assert in the next PR.

Reply:
Ok